### PR TITLE
Rework the compression library search

### PR DESCRIPTION
**Pull request by byrnHDF**
Added default names for search use.

### CODE REVIEW

**Comment by byrnHDF** [comment]
The search path for libs can be controlled by CMake system variables - see https://cmake.org/cmake/help/latest/command/find_package.html

File: CMakeFilters.cmake [hunk 1]
```diff
@@ -92,7 +98,8 @@ if (HDF5_ENABLE_ZLIB_SUPPORT)
       else ()
         set(ZLIB_SEACH_TYPE shared)
       endif ()
-      find_package (ZLIB NAMES ${PACKAGE_NAME} COMPONENTS ${ZLIB_SEACH_TYPE})
+      # Search pure Config mode first
+      find_package (ZLIB NAMES ${PACKAGE_NAME} OPTIONAL_COMPONENTS ${ZLIB_SEACH_TYPE})
```
Comment (jhendersonHDF) [inline CMakeFilters.cmake]: `ZLIB_SEACH_TYPE` -> `ZLIB_SEARCH_TYPE`

File: CMakeFilters.cmake [hunk 2]
```diff
@@ -149,6 +156,9 @@ endif ()
 # Option for SzLib support
 #-----------------------------------------------------------------------------
 set(H5_SZIP_FOUND FALSE)
+if(NOT DEFINED LIBAEC_PACKAGE_NAME)
+  set(LIBAEC_PACKAGE_NAME "libaec")
+endif ()
 option (HDF5_ENABLE_SZIP_SUPPORT "Use SZip Filter" ON)
```
Comment (lrknox) [inline CMakeFilters.cmake]:
```suggestion
option (HDF5_ENABLE_SZIP_SUPPORT "Use SZip Filter" OFF)
```

**Review by lrknox — DISMISSED** [review]

**Review by lrknox — DISMISSED** [review]

**Comment by lrknox** [comment]
This command successfully downloads and configures zlib szip and hdf5_plugins from their GitHub repos along with HDF5 develop:  
cmake -C ../hdf5-2.0.0/config/cmake/cacheinit.cmake -G "Unix Makefiles" -DHDF5_ALLOW_EXTERNAL_SUPPORT:STRING="GIT" -DZLIB_USE_LOCALCONTENT=OFF -DLIBAEC_USE_LOCALCONTENT=OFF -DHDF5_ENABLE_ROS3_VFD=ON -DHDF5_ENABLE_ZLIB_SUPPORT=ON -DHDF5_ENABLE_SZIP_SUPPORT=ON -DHDF5_ENABLE_PLUGIN_SUPPORT=ON -DCMAKE_BUILD_TYPE:STRING=Debug --log-level=VERBOSE ../hdf5-2.0.0

This one fails to download hdf5_plugins.tar.gz.  Zlib and szip tar.gz files are downloaded:
cmake -C ../hdf5-2.0.0/config/cmake/cacheinit.cmake -G "Unix Makefiles" -DHDF5_ALLOW_EXTERNAL_SUPPORT:STRING="TGZ" -DZLIB_USE_LOCALCONTENT=OFF -DPLUGIN_USE_LOCALCONTENT=OFF -DLIBAEC_USE_LOCALCONTENT=OFF -DHDF5_ENABLE_ROS3_VFD=ON -DHDF5_ENABLE_ZLIB_SUPPORT=ON -DHDF5_ENABLE_SZIP_SUPPORT=ON -DHDF5_ENABLE_PLUGIN_SUPPORT=ON -DCMAKE_BUILD_TYPE:STRING=Debug --log-level=VERBOSE ../hdf5-2.0.0

This works:  wget https://github.com/HDFGroup/hdf5_plugins/releases/download/snapshot/hdf5_plugins-master.tar.gz
This fails:  https://github.com/HDFGroup/hdf5_plugins/releases/download/snapshots/hdf5_plugins-master.tar.gz

It appears the CMakePresets.json is correct, but cacheinit.cmake and INSTALL_CMake.txt need the final 's' removed from 'snapshots':
[lrknox@jelly hdf5-2.0.0]$ grep -r hdf5_plugins * | grep download
CMakePresets.json:        "PLUGIN_TGZ_ORIGPATH": {"type": "STRING", "value": "https://github.com/HDFGroup/hdf5_plugins/releases/download/snapshot"},
config/cmake/cacheinit.cmake:set (PLUGIN_TGZ_ORIGPATH "https://github.com/HDFGroup/hdf5_plugins/releases/download/snapshots" CACHE STRING "Use PLUGINS from original location" FORCE)
release_docs/INSTALL_CMake.txt:            set (PLUGIN_TGZ_ORIGPATH "https://github.com/HDFGroup/hdf5_plugins/releases/download/snapshots" CACHE STRING "Use PLUGINS from original location" FORCE)

File: release_docs/INSTALL_CMake.txt [hunk 3]
```diff
@@ -68,7 +68,7 @@ Download from https://github.com/HDFGroup/hdf5/blob/develop/config/cmake/scripts
 External compression plugin libraries from https://github.com/HDFGroup/hdf5_plugins:
     hdf5_plugins.tar.gz
 
-External compression szip and zlib libraries:
+External compression szip and zlib libraries:0
```
Comment (jhendersonHDF) [inline release_docs/INSTALL_CMake.txt]: Accidental?

File: release_docs/RELEASE.txt [hunk 4]
```diff
@@ -50,7 +50,7 @@ New Features
     - Renamed the option: HDF5_ENABLE_Z_LIB_SUPPORT
 
       The option has been renamed to HDF5_ENABLE_ZLIB_SUPPORT to be consistent
-      with the naming of other options.
+      with the naming of other options. Also, the option defaults to OFF.
```
Comment (jhendersonHDF) [inline release_docs/RELEASE.txt]: Definitely want to make sure this is very well called out for the next release

File: release_docs/INSTALL_CMake.txt (hunk 3 above)
Comment (byrnHDF) [reply release_docs/INSTALL_CMake.txt]: Most likely yes.

File: release_docs/RELEASE.txt (hunk 4 above)
Comment (byrnHDF) [reply release_docs/RELEASE.txt]: Absolutely

**Review by lrknox — DISMISSED** [review]

File: release_docs/NEWSLETTER.txt [hunk 5]
```diff
@@ -3,7 +3,13 @@ HDF5 version 2.0.0 currently under development
 Features included for the next major release:
 ----------------------------------------------------------------------------
 
-*
+      ************ Renamed the option: HDF5_ENABLE_Z_LIB_SUPPORT ************
```
Comment (mattjala) [inline release_docs/NEWSLETTER.txt:6]: Could we break this into two items, with one indicating the rename and one indicating that it now defaults to off? We'd want someone skimming the headlines to catch both of these

File: release_docs/RELEASE.txt [hunk 6]
```diff
@@ -47,10 +47,12 @@ New Features
 
     Configuration:
     -------------
-    - Renamed the option: HDF5_ENABLE_Z_LIB_SUPPORT
+    - ************ Renamed the option: HDF5_ENABLE_Z_LIB_SUPPORT ************
 
       The option has been renamed to HDF5_ENABLE_ZLIB_SUPPORT to be consistent
       with the naming of other options.
+      *** Also, the option defaults to OFF. This requires the user to explicitly ***
```
Comment (mattjala) [inline release_docs/RELEASE.txt:54]: Same idea about breaking this into two items

**Comment by brtnfld** [comment]
I suggest we transition from using txt files to markdown, enabling us to highlight important information more effectively by the use of:

https://github.blog/changelog/2023-12-14-new-markdown-extension-alerts-provide-distinctive-styling-for-significant-content/

**Review by lrknox — DISMISSED** [review]

**Review by lrknox — DISMISSED** [review]

**Review by jhendersonHDF — DISMISSED** [review]

**Review by lrknox — APPROVED** [review]

**Pull request merged**